api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java

```diff
@@ -62,7 +62,20 @@
  * All Built In Converters have a {@code javax.annotation.Priority} of 1
  * A Converter should handle null values returning either null or a valid Object of the specified type.
  *
- *  *
+ * <h3>Array Converters</h3>
```

Comment:
I like the idea, but I think for this to be concrete we have to update the spec as well.  The TCK should be updated to test this for all of the current types.

If we are going to include custom converters, we need to enumerate what the separators are.  We would also need to a way to escape the values.  For instance if you took your change and did instead:

```
tck.config.test.javaconfig.converter.array.pizza=large:ham,pineapple,medium:chicken,small:pepperoni
```

I'm trying to convey that the large pizza should have both ham and pineapple toppings.  There should be a way to make those two go together.

Reply:
@johnament I like your comments here:). Yes, I will change the spec about the delimiter (,) and escape character(\).

Reply:
More commits to come to this one as I have done this to demonstrate to ConfigJsr EG.

Reply:
I agree with @johnament - the behavior of the converter should be specified in the spec because config sources can be provided also by the application and it should be clear how the array converter would process string values. More concretely, things like separators, escape sequence for a separator and converting string chunks into Java objects should be covered.

Reply:
yep. agreed. More commits to come.